src/utils/api-request.ts

```diff
@@ -398,15 +386,139 @@ export class HttpClient {
   }
 }
 
+export class HttpClient extends RequestClient {
+
+  constructor(retry?: RetryConfig | null) {
+    super(retry)
+  }
+
+  /**
+   * Sends an HTTP request to a remote server. If the server responds with a successful response (2xx), the returned
+   * promise resolves with an RequestResponse. If the server responds with an error (3xx, 4xx, 5xx), the promise rejects
+   * with an RequestResponseError. In case of all other errors, the promise rejects with a FirebaseAppError. If a
+   * request fails due to a low-level network error, transparently retries the request once before rejecting the
```

Comment:
Let's provide a subject noun here -- what entity retries?  The server/backend/promise?